### PR TITLE
Feature#23 tests and validations for company branch model

### LINKED ISSUE #23: Tests and Validations for the Company Branch Model
# REQUIREMENTS
Validate form inputs to create company branches and test ones.
Add and config shoulda-matchers to avoid that the tests are much longer, more complex and prone to errors.
# Acceptance Criteria
- [x] validates presence of company branch attributes
- [x] tests company branch model
- [x] tests company branch associations

### PR DESCRIPTION
**Pull request by EusebioAjas**
# Description
 - I've added presence of attributes in CompanyBranch model for testing all attributes and associations.
 - These changes complete the new CompanyBranch view and validate form inputs to create ones. See [issue#23](https://github.com/BC-MAR-21-ROR/kata-07-rejoj-checador-equipo-03-k07/issues/23) for more information.
 # Testing
  - I used shoulda-matchers for testing presence of attributes and associations. See [shoulda-matchers](https://github.com/thoughtbot/shoulda-matchers) for more information.
  
You can run the tests writing:
 `rspec spec/models/company_branch_spec.rb`

company_branch_spec

  
  ## Screenshots 
new_branch

### CODE REVIEW

File: spec/support/factory_bot.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# frozen_string_literal: true
+
+RSpec.configure do |config|
+  config.include FactoryBot::Syntax::Methods
```
Comment (Arkamis) [inline spec/support/factory_bot.rb:4]: Just 1 question here. Where do you use factorybot gem, or why you added it?

**Review by Arkamis — DISMISSED** [review]
LGTM!, thanks for the simplification in the commits

File: spec/support/factory_bot.rb (hunk 1 above)
Comment (EusebioAjas) [reply spec/support/factory_bot.rb:4]: The gem was previously added in this PR: [ Feature#16 create test data](https://github.com/BC-MAR-21-ROR/kata-07-rejoj-checador-equipo-03-k07/pull/21)

Comment (Arkamis) [reply spec/support/factory_bot.rb:4]: oh gotcha

File: Gemfile [hunk 2]
```diff
@@ -64,6 +64,8 @@ end
 group :test do
   # Adds support for Capybara system testing and selenium driver
   gem 'capybara', '>= 3.26'
+  gem 'database_cleaner-active_record', '~> 2.0'
+  gem 'shoulda-matchers', '~> 4.5', '>= 4.5.1'
```
Comment (jnegrete15) [inline Gemfile]: Nice gem!, it looks useful

**Review by jnegrete15 — APPROVED** [review]

**Review by jnegrete15 — DISMISSED** [review]

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -9,10 +9,9 @@ def output_path
   end
 end
 SimpleCov.minimum_coverage 65
-SimpleCov.formatter = SimpleCov::Formatter::MultiFormatter[
-  SimpleCov::Formatter::HTMLFormatter,
-	SimpleCov::Formatter::BadgeFormatter,
-]
+SimpleCov.formatter = SimpleCov::Formatter::MultiFormatter.new \
```
Comment (anxhe) [inline spec/spec_helper.rb:12]: Thanks for amending this

**Review by anxhe — APPROVED** [review]
Good job, we now can test better

**Review by RobertVel666 — APPROVED** [review]
meets the criteria great work

**Pull request merged**